### PR TITLE
Add clone command

### PR DESCRIPTION
**Pull request by HeDo88TH**
Adds the clone command as stated in https://github.com/DroneDB/DroneDB/issues/22#issuecomment-768372959

### CODE REVIEW

File: src/cmd/clone.cpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+namespace cmd {
+void Clone::setOptions(cxxopts::Options& opts) {
+    // clang-format off
+        opts
+			.positional_help("[args]")
+			.custom_help("clone (tag|url)")
+			.add_options()
+			("t,target", "Repository tag or full url", cxxopts::value<std::string>())
+			("f,folder", "Target folder", cxxopts::value<std::string>());
```
Comment (pierotofy) [inline src/cmd/clone.cpp:23]: This should be optional: 

```
ddb clone pierotofy/sheffield_park
```
 --> clone into `sheffield_park` directory (if it doesn't exist, error out if directory already exists).

**Comment by HeDo88TH** [comment]
It does not build with this error:

```
clone.obj : error LNK2019: riferimento al simbolo esterno "public: __cdecl homer6::Url::Url(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (??0Url@homer6@@QEAA@AEBV?$basic_string@DU?$ch 
ar_traits@D@std@@V?$allocator@D@2@@std@@@Z) non risolto nella funzione "private: struct cmd::TargetInfo __cdecl cmd::Clone::getCloneTarget(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > &)" (?get 
CloneTarget@Clone@cmd@@AEAA?AUTargetInfo@2@AEAV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@Z) [C:\Users\ldile\source\repos\DroneDB\build\ddbcmd.vcxproj]
clone.obj : error LNK2019: riferimento al simbolo esterno "public: class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl homer6::Url::getPath(void)const " (?getPath@Url@homer6@@QEBA?AV?$basic_st 
ring@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@XZ) non risolto nella funzione "private: struct cmd::TargetInfo __cdecl cmd::Clone::getCloneTarget(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char>  
> &)" (?getCloneTarget@Clone@cmd@@AEAA?AUTargetInfo@2@AEAV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@Z) [C:\Users\ldile\source\repos\DroneDB\build\ddbcmd.vcxproj]
C:\Users\ldile\source\repos\DroneDB\build\Release\ddbcmd.exe : fatal error LNK1120: 2 esterni non risolti [C:\Users\ldile\source\repos\DroneDB\build\ddbcmd.vcxproj]
```

I checked the file `vendor/url/CMakeLists.txt` and it seems fine. Why is it not finding the url library? @pierotofy

File: src/cmd/clone.cpp (hunk 1 above)
Comment (HeDo88TH) [reply src/cmd/clone.cpp:25]: Addressed in https://github.com/DroneDB/DroneDB/pull/190/commits/673801c6c48772d8e380c16fda9cf2bcba0724ec

File: src/cmd/clone.cpp [hunk 2]
```diff
@@ -34,20 +36,71 @@ std::string Clone::extendedDescription() {
     return "\r\n\r\nClones a repository into a newly created directory.";
 }
 
+TargetInfo Clone::getCloneTarget(std::string& target) {
+    TargetInfo info;
+
+    if (target.find("http://") == std::string::npos &&
+        target.find("https://") == std::string::npos) {
+        // We assume it is not a link
+        const auto needle = target.find('/');
```
Comment (pierotofy) [inline src/cmd/clone.cpp:45]: Use `RegistryUtils::parseTag` instead; it covers all use cases.

File: src/cmd/clone.h [hunk 3]
```diff
@@ -9,7 +9,17 @@
 
 namespace cmd {
 
+struct TargetInfo {
```
Comment (pierotofy) [inline src/cmd/clone.h:12]: Duplicate of `struct TagComponents`. :)

Comment (HeDo88TH) [reply src/cmd/clone.h:12]: Fixed

File: src/registry.cpp [hunk 4]
```diff
@@ -162,7 +162,7 @@ DDB_DLL void Registry::clone(const std::string &organization,
 
     std::filesystem::remove(tempFile);
 
-    std::cout << "Removed temp file" << std::endl
+    std::cout << "Removed temp file" << std::endl;
```
Comment (pierotofy) [inline src/registry.cpp:165]: Remember not to pollute stdout!

Comment (HeDo88TH) [reply src/registry.cpp:165]: It's still a draft 👍

File: src/registry.cpp [hunk 5]
```diff
@@ -116,9 +116,28 @@ bool Registry::logout() {
...
 }
 
+void smartDisplay(std::ostream& stream, double number) {
+
+    stream << std::setprecision(4);
+
+    if (number < 1024) {
+        //
+        stream << number << " bytes";
+    } else if (number < 1048576)
```
Comment (pierotofy) [inline src/registry.cpp:126]: Use `io::bytesToHuman(number)` :+1:

Comment (HeDo88TH) [reply src/registry.cpp]: Done in https://github.com/DroneDB/DroneDB/pull/190/commits/a3950f1dd6e847d0cb56eafdbe2d26ed32ad5903

**Comment by pierotofy** [comment]
The `exiv2` submodule was mistakenly updated. I've set it to track the proper branch.

**Comment by pierotofy** [comment]
URL parsing was broken for tags such as `http://localhost:5000/admin/test`. It's fixed now.

**Comment by pierotofy** [comment]
I/O operations should be caught. (Fixed)

**Pull request merged**